src/measure_theory/interval_integral.lean

```diff
@@ -370,39 +370,110 @@ lemma integral_smul_measure (c : ℝ≥0∞) :
   ∫ x in a..b, f x ∂(c • μ) = c.to_real • ∫ x in a..b, f x ∂μ :=
 by simp only [interval_integral, measure.restrict_smul, integral_smul_measure, smul_sub]
 
-lemma integral_comp_add_right {a b : ℝ} (c : ℝ) {f : ℝ → E} (hfm : ae_measurable f) :
-  ∫ x in a..b, f (x + c) = ∫ x in a+c..b+c, f x :=
-have A : ae_measurable f (measure.map (λ x, x + c) volume), by rwa [real.map_volume_add_right],
-calc ∫ x in a..b, f (x + c) = ∫ x in a+c..b+c, f x ∂(measure.map (λ x, x + c) volume) :
-  by simp only [interval_integral, set_integral_map measurable_set_Ioc A (measurable_add_const _),
-    preimage_add_const_Ioc, add_sub_cancel]
-... = ∫ x in a+c..b+c, f x : by rw [real.map_volume_add_right]
+end basic
+
+section comp
 
-lemma integral_comp_mul_right {a b c : ℝ} {f : ℝ → E} (hc : 0 < c) (hfm : ae_measurable f) :
+variables {a b c : ℝ} (f : ℝ → E)
+
+lemma integral_comp_mul_right_of_pos (hc : 0 < c) :
   ∫ x in a..b, f (x * c) = c⁻¹ • ∫ x in a*c..b*c, f x :=
 begin
-  have A : ae_measurable f (measure.map (λ (x : ℝ), x*c) volume),
-    by { rw real.map_volume_mul_right (ne_of_gt hc), exact hfm.smul_measure _ },
+  have A : closed_embedding (λ (x : ℝ), x * c) :=
+    (homeomorph.mul_right' c (ne_of_gt hc)).closed_embedding,
   conv_rhs { rw [← real.smul_map_volume_mul_right (ne_of_gt hc)] },
   rw [integral_smul_measure],
-  simp only [interval_integral, set_integral_map measurable_set_Ioc A (measurable_mul_const _),
-    hc, preimage_mul_const_Ioc, mul_div_cancel _ (ne_of_gt hc), abs_of_pos,
+  simp only [interval_integral, hc, preimage_mul_const_Ioc, mul_div_cancel _ (ne_of_gt hc),
+    abs_of_pos, set_integral_map_of_closed_embedding measurable_set_Ioc A,
     ennreal.to_real_of_real (le_of_lt hc), inv_smul_smul' (ne_of_gt hc)],
```

Comment:
```suggestion
  have A : closed_embedding (λ (x : ℝ), x * c) :=
    (homeomorph.mul_right' c hc.ne').closed_embedding,
  conv_rhs { rw [← real.smul_map_volume_mul_right hc.ne'] },
  rw [integral_smul_measure],
  simp only [interval_integral, hc, preimage_mul_const_Ioc, mul_div_cancel _ hc.ne',
    abs_of_pos, set_integral_map_of_closed_embedding measurable_set_Ioc A,
    ennreal.to_real_of_real hc.le, inv_smul_smul' hc.ne'],
```
Changing all the `ne_of_gt hc` to `hc.ne'`, and all the `le_of_lt hc` to `hc.le` ... a cosmetic change which I think helps with readability.

Reply:
I very much agree. I'll make these changes.